### DISCUSSION TITLE
CS8603 Warning When Using Multiple Fields in Switch Expression

### DISCUSSION THREAD

**Discussion by KalumeC** [dotnet/csharplang#8513] [Q&A] [answered]
Hello

```csharp
private static object GetSortingValue(SomeClass obj, SomeEnum field)
{
    return field switch
    {
        SomeEnum.Field1 => obj.Field1,
        SomeEnum.Field2 => obj.Field2,
        SomeEnum.Field3 => obj.Field3,
        SomeEnum.Field4 => obj.Field4,
        SomeEnum.Field5 => obj.Field5,
        SomeEnum.Field6 => obj.Field6,
        _ => throw new ArgumentOutOfRangeException(nameof(field))
    };
}
```

This code triggers a CS8603 warning, requiring the return type to be nullable. However, if the `switch` expression contains only two arms one pattern and one discard, the warning does not occur.

Can someone explain why this warning happens and suggest the best way to handle it?

Thank you.

**Comment by KennethHoff** [comment, marked as answer]
We would need to know what the fields of `SomeClass` are

**Reply by KalumeC** [reply]
Well I was making a small reproduction and noticed a nullable pragma hidden inside a region, the field were indeed nullable! I should have looked better.

Thanks for the answer
